controllers/openstackdataplanenodeset_controller.go

```diff
@@ -687,13 +659,15 @@ func (r *OpenStackDataPlaneNodeSetReconciler) SetupWithManager(mgr ctrl.Manager)
 		Owns(&infranetworkv1.DNSData{}).
 		Owns(&corev1.Secret{}).
 		Watches(&infranetworkv1.DNSMasq{},
-			dnsMasqWatcher).
+			genericWatcher).
 		Watches(&dataplanev1.OpenStackDataPlaneDeployment{},
 			deploymentWatcher).
 		Watches(&corev1.ConfigMap{},
 			varsFromWatcher, builder.WithPredicates(predicate.ResourceVersionChangedPredicate{})).
 		Watches(&corev1.Secret{},
 			varsFromWatcher, builder.WithPredicates(predicate.ResourceVersionChangedPredicate{})).
+		Watches(&openstackv1.OpenStackVersion{},
+			genericWatcher).
```

Comment:
TODO: this watcher seems to be causing infinite reconcile loops. Not sure why yet as I don't see the OpenStackVersion resource actually changing.

Reply:
instead of a 'genericWatcher' function you may try something more specific to OpenStackVersion where you can return false if no update is required.

Reply:
I tried that and it doesn't make a difference. The cause is actually because when I run openstack-operator locally, it's looping endlessly as well causing OpenStackVersion to constantly reconcile, and thus the NodeSet. That is probably something specific to my environment with the control plane I have deployed, so I'll consider this one a non-issue for now.